cms/web/practice/js/task.js

```diff
@@ -220,4 +217,18 @@ angular.module('pws.task', [])
     $scope.showDetails = function(id) {
       subsDatabase.subDetails(id);
     };
+  })
+  .directive('pdf', function() {
+    return {
+      restrict: 'E',
+      link: function(scope, element, attrs) {
+        var hasBuiltInPdf = !("ActiveXObject" in window) && !/iPhone|iPod|Android|BlackBerry|Opera Mini|Phone|Mobile/i.test(navigator.userAgent);
```

Comment:
`"ActiveXObject" in window` ritorna true se è internet explorer (nessuna versione di ie lo ha nativo).
Il regex invece serve per capire se è un dispositivo mobile o meno, non ho inserito tanti nomi ma credo bene o male bastino quelli e non è presente 'iPad' perché pare che il suo safari abbia un visualizzatore integrato. in alternativa si potrebbe anche usare `if(typeof window.orientation !== 'undefined'`){...} che è però meno flessibile e rischia di funzionare anche [quando non dovrebbe](http://stackoverflow.com/a/14301832).